fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/MetadataManager.java

```diff
@@ -343,6 +347,23 @@ public void createPartition(
                             partition.getPartitionQualifiedName(), tablePath));
         }
 
+        try {
+            int partitionNumber = zookeeperClient.getPartitionNumber(tablePath);
+            int configMaxSize = conf.get(ConfigOptions.MAX_PARTITION_NUM);
+            if (partitionNumber + 1 > configMaxSize) {
+                throw new TooManyPartitionsException(
+                        String.format(
+                                "Exceed the maximum number of partitions for table %s", tablePath));
+            }
+        } catch (TooManyPartitionsException e) {
+            throw e;
+        } catch (Exception e) {
+            LOG.error(
+                    "Get the number of partition from zookeeper failed for table [{}]",
+                    tablePath,
+                    e);
```

Comment:
We should throw the exception to allow users (CoordinatorService) know the partition creation failed (like how `createTable` does). The following exception catch should also be removed. In the `AutoPartitionManager`, we should catch the exception and log error messages.